src/docs/services/relay/projectconfig-versioning.mdx

```diff
@@ -0,0 +1,138 @@
+
+- In Sentry, this is handled by an [early return](https://github.com/getsentry/sentry/blob/b1434710fff9d8f90c99668bcc958ed1ee52b66e/src/sentry/relay/config.py#L165-L167) to stop populating the config object.
+- In Relay, this is implemented by having two separate types for trusted vs restricted project config, either `ProjectConfig` (consumed by trusted Relays) or `LimitedProjectConfig` (consumed by untrusted Relays).
+
+For as long as your new feature is in development, it's recommended to emit new/different fields only for trusted Relays. 
+
+## Defining New Structs in Relay
+
+- use `#[serde(rename_all = "camelCase")]`, even if all fields are single words.
+- never specify `deny_unknown_fields`
+
+New structs are allowed to contain mandatory fields without a default.
+
+## Defining New Enums
+
+There are two kinds of enums:
+
+- Plain string enumerations. The string value determines the variant of the enumeration.
+- Internally tagged enumerations. These are structures (JSON objects) with a discriminator column that indicates the variant.
+- Untagged enumerations. When the type of the field can differ, this allows to serialize the first matching variant. This is used, for example, if a field can either be a string or a number.
+
+Unlike structs, unknown variants are not implicitly ignored by Relay. When defining an enumeration, *always* add a variant to catch unknown variants using `#[serde(other)]`. For example:
+
+```rust
+#[derive(Deserialize, Serialize)]
+#[serde(tag = "type", rename_all = "camelCase")]
+enum BreakdownConfig {
+    SpanOperations { ... },
+    #[serde(other)]
+    Unsupported,
+}
+```
+
+Before adding a new variant in Sentry, verify that Relay implements this correctly. If this is not the case, proceed by replacing the field just like when changing a fields type. Never add variants to an enum that does not have an unknown variant.
+
+It is acceptable that Relay does not forward unknown enum variants by dropping the original value, as it is done with the above approach.
+
+## Adding Top-Level Fields
+
+In case you're developing a new feature, you likely want to add completely new data to the project config and not touch existing structs. Here's a small checklist that applies in addition to everything else:
+
+- Verify that the field is in `camelCase`.
+- Ensure the field name has not been used in a past version of Relay. `git log -G field_name` is a simple way to search git history for strings that occur in diffs.
+- Make sure your field is either optional or has a default
+- Do not serialize when the field is `None` or default via `serde(skip_serializing_if)`.
+- Use `ErrorBoundary` around the field type for as long as your new feature is pre-GA and can change its definition. This makes sure that if the field definition is malformed, Relay contains the error and continues to parse the rest of the project config.
+- Do not use `usize`.
```

Comment:
```suggestion
- Do not use types with platform specific behavior, such as `usize`.
```

Not sure if there are any other types we'd ever consider adding there, but it's better to show clear intent.